zipline/assets/assets.py

```diff
@@ -703,6 +704,10 @@ def lookup_symbol(self, symbol, as_of_date, fuzzy=False):
             there are multiple candidates for the given ``symbol`` on the
             ``as_of_date``.
         """
+        if symbol is None:
+            raise TypeError("Cannot lookup symbol of type NoneType for "
+                            "as of date %s" % as_of_date)
```

Comment:
I guess my thinking was that fuzzy or not would have been passed to the method, and that would be something known by the user from the get-go. The symbol is unhelpful to pass back, since it's None, but that a user might want to identify _where_ the symbol was None in their data, the only reasonable identifier for which was the as_of_date.